repo2docker/buildpacks/base.py

```diff
@@ -271,16 +271,54 @@ def stencila_manifest_dir(self):
         # ${STENCILA_ARCHIVE_DIR}/${STENCILA_ARCHIVE}/manifest.xml
 
         self._stencila_manifest_dir = None
+
+        if os.path.exists('binder'):
+            return self._stencila_manifest_dir
+
         for root, dirs, files in os.walk("."):
             if "manifest.xml" in files:
+                self.log.debug("Found a manifest.xml at %s", root)
                 self._stencila_manifest_dir = root.split(os.path.sep, 1)[1]
                 self.log.info(
-                    "Found stencila manifest.xml in %s",
+                    "Using stencila manifest.xml in %s",
                     self._stencila_manifest_dir,
                 )
                 break
         return self._stencila_manifest_dir
 
+    @property
+    def stencila_manifest_contexts(self):
```

Comment:
Could we rename this to `stencila_contexts`? Makes it a bit easier to read I think and helps with line length later on.

Or are there contexts that don't come from manifests?

Reply:
Well, actually the contexts do not come from the manifests, but from the document files :see_no_evil:  - so good point, renamed.